lib/ex_toolkit/map.ex

```diff
@@ -0,0 +1,45 @@
+defmodule ExToolkit.Map do
+  @moduledoc """
+  This module provides utilities for working with maps.
+  """
+
+  @doc """
+  Recursively converts a struct into a plain map, transforming all nested structs,
+  maps, and lists along the way.
+
+  Only accepts a struct at the top level. If a non-struct is passed, it will raise a `FunctionClauseError`.
+
+  Nested values (inside maps, lists, or other structs) will be traversed, and any structs found will also be converted to maps.
```

Comment:
This is a bit repeating the first line. I think you should replace it with something like ```If you don't want to do a deep recursive conversion you should just use `Map.from_struct/1`.```